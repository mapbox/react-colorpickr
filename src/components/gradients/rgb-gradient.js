'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import themeable from 'react-themeable';

class RGBGradient extends React.Component {
  static propTypes = {
    theme: PropTypes.object.isRequired,
    active: PropTypes.bool.isRequired,
    color: PropTypes.string.isRequired,
    opacityLow: PropTypes.object.isRequired,
    opacityHigh: PropTypes.object.isRequired
  };

  render() {
    const theme = themeable(this.props.theme);
    const { active, color, opacityLow, opacityHigh } = this.props;
    if (!active) return <noscript />;
    return (
      <div>
        <div {...theme(1, 'gradient', `gradient${color.toUpperCase()}High`)} style={opacityHigh} />
        <div {...theme(2, 'gradient', `gradient${color.toUpperCase()}Low`)} style={opacityLow} />
      </div>
    );
  }
}

export default RGBGradient;
