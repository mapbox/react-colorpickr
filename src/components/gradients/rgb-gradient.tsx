import React from 'react';
import PropTypes from 'prop-types';
import themeable from 'react-themeable';
import { autokey } from '../../autokey.ts';

function RGBGradient({ theme, active, color, opacityLow, opacityHigh }) {
  const themer = autokey(themeable(theme));
  if (!active) return <noscript />;
  return (
    <>
      <div
        {...themer('gradient', `gradient${color.toUpperCase()}High`)}
        style={{ opacity: opacityHigh }}
      />
      <div
        {...themer('gradient', `gradient${color.toUpperCase()}Low`)}
        style={{ opacity: opacityLow }}
      />
    </>
  );
}

RGBGradient.propTypes = {
  theme: PropTypes.object.isRequired,
  active: PropTypes.bool.isRequired,
  color: PropTypes.string.isRequired,
  opacityLow: PropTypes.number.isRequired,
  opacityHigh: PropTypes.number.isRequired
};

export { RGBGradient };
