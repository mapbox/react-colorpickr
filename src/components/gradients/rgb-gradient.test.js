import React from 'react';
import { render } from '@testing-library/react';
import { RGBGradient } from './rgb-gradient';

describe('RGBGradient', () => {
  test('renders', () => {
    const props = {
      active: true,
      color: 'r',
      theme: {},
      opacityLow: 0,
      opacityHigh: 0
    };

    const { baseElement } = render(<RGBGradient {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('renders with opacity', () => {
    const props = {
      active: false,
      color: 'r',
      theme: {},
      opacityLow: 0.5,
      opacityHigh: 0.6
    };

    const { baseElement } = render(<RGBGradient {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
