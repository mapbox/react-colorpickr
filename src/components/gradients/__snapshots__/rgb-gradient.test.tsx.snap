// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RGBGradient renders 1`] = `
<body>
  <div>
    <div
      style="opacity: 0;"
    />
    <div
      style="opacity: 0;"
    />
  </div>
</body>
`;

exports[`RGBGradient renders with opacity 1`] = `
<body>
  <div>
    <noscript />
  </div>
</body>
`;
